app/utils/callback/lava.py

```diff
@@ -638,6 +488,183 @@ def store_rootfs_info(self, file_name):
             utils.LOG.warn("ValueError: {}".format(e))
 
 
+def store_artifacts(metadata, job_data, log):
+    store_test_log(metadata, log)
+    store_lava_json(metadata, job_data)
+
+
+def store_lava_json(metadata, job_data):
+    """ Save the json LAVA v2 callback object
+
+    Save LAVA v2 callback data as json file.
+    """
+
+    file_name = "-".join(["lava-json", metadata[models.DEVICE_TYPE_KEY]])
+    file_name = ".".join([file_name, "json"])
+
+    dir_path = metadata[models.DIRECTORY_PATH]
+
+    utils.LOG.info("Saving LAVA v2 callback file {} data in {}".format(
+        file_name,
+        dir_path))
+
+    file_path = os.path.join(dir_path, file_name)
+
+    # Removing the token
+    job_data.pop("token", None)
+
+    # Add extra information
+    job_data["lab_name"] = metadata.get("lab_name")
+    job_data["version"] = metadata.get("version")
+    job_data["boot_log_html"] = metadata.get("boot_log_html")
+
+    if not os.path.isdir(dir_path):
+        try:
+            os.makedirs(dir_path)
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise e
```

Comment:
Seems worth factoring out into a utility function.

Reply:
extracted to `utils.make_path()` function